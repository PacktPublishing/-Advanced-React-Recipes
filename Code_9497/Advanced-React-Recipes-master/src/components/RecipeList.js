import React from 'react';
import PropTypes from 'prop-types';
import { CSSTransitionGroup } from 'react-transition-group';
import RecipeListItem from './RecipeListItem';

const RecipeList = ({ recipes, favorites, className, style, ...props }) => (
  <div className={className} style={style}>
    <ul className="list-reset">
      <CSSTransitionGroup
        transitionName="slideOutRight"
        transitionLeaveTimeout={300}
        transitionEnter={false}
      >
        {recipes.map(recipe => (
          <RecipeListItem
            key={recipe.id}
            recipe={recipe}
            favorited={favorites.includes(recipe.id)}
            {...props}
          />
        ))}
      </CSSTransitionGroup>
    </ul>
  </div>
);

RecipeList.propTypes = {
  recipes: PropTypes.array,
  favorites: PropTypes.array,
  onClick: PropTypes.func,
  className: PropTypes.string,
  style: PropTypes.object,
};

RecipeList.defaultProps = {
  recipes: [],
  favorites: [],
};

export default RecipeList;
