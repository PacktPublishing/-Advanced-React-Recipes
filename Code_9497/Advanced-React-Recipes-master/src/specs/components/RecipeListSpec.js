import React from 'react';
import renderer from 'react-test-renderer';
import RecipeList from '../../components/RecipeList';

const testRecipes = [
  {
    id: 1,
    name: 'Test recipe 1',
    category: 'Dessert',
  },
  {
    id: 2,
    name: 'Test recipe 2',
    category: 'Meat',
  },
];

describe('<RecipeList />', () => {
  let component;
  let tree;

  test('Should not break if no recipes passed', () => {
    expect(() => renderer.create(<RecipeList />)).not.toThrow();
  });

  test('Should render passed recipes', () => {
    component = renderer.create(<RecipeList recipes={testRecipes} />);
    tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should render favorited states', () => {
    component = renderer.create(
      <RecipeList recipes={testRecipes} favorites={[1]} />,
    );
    tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
