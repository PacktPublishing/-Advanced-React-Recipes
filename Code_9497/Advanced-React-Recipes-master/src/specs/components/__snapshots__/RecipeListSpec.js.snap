// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeList /> Should render favorited states 1`] = `
<div
  className={undefined}
  style={undefined}
>
  <ul
    className="list-reset"
  >
    <span>
      <li
        className="py2 border-bottom border-bottom-dashed pointer nowrap"
        onClick={[Function]}
      >
        <div
          className="inline-block animated"
          onClick={[Function]}
          role="button"
        >
          ✅
        </div>
        <span>
          Test recipe 1
        </span>
        <span>
          Dessert
        </span>
      </li>
      <li
        className="py2 border-bottom border-bottom-dashed pointer nowrap"
        onClick={[Function]}
      >
        <div
          className="inline-block animated"
          onClick={[Function]}
          role="button"
        >
          ⬜️
        </div>
        <span>
          Test recipe 2
        </span>
        <span>
          Meat
        </span>
      </li>
    </span>
  </ul>
</div>
`;

exports[`<RecipeList /> Should render passed recipes 1`] = `
<div
  className={undefined}
  style={undefined}
>
  <ul
    className="list-reset"
  >
    <span>
      <li
        className="py2 border-bottom border-bottom-dashed pointer nowrap"
        onClick={[Function]}
      >
        <div
          className="inline-block animated"
          onClick={[Function]}
          role="button"
        >
          ⬜️
        </div>
        <span>
          Test recipe 1
        </span>
        <span>
          Dessert
        </span>
      </li>
      <li
        className="py2 border-bottom border-bottom-dashed pointer nowrap"
        onClick={[Function]}
      >
        <div
          className="inline-block animated"
          onClick={[Function]}
          role="button"
        >
          ⬜️
        </div>
        <span>
          Test recipe 2
        </span>
        <span>
          Meat
        </span>
      </li>
    </span>
  </ul>
</div>
`;
